src/pybind/mgr/cephadm/inventory.py

```diff
@@ -1699,6 +1764,294 @@ def agent_config_successfully_delivered(self, daemon_spec: CephadmDaemonDeploySp
+        return NotImplemented
+
+    def to_json(self) -> Dict[str, Union[str, bool]]:
+        return {
+            'cert': self.cert,
+            'user_made': self.user_made
+        }
+
+    @classmethod
+    def from_json(cls, data: Dict[str, Union[str, bool]]) -> 'Cert':
+        if 'cert' not in data:
+            return cls()
+        cert = data['cert']
+        if not isinstance(cert, str):
+            raise OrchestratorError('Tried to make Cert object with non-string cert')
+        if any(k not in ['cert', 'user_made'] for k in data.keys()):
+            raise OrchestratorError(f'Got unknown field for PrivKey object. Fields: {data.keys()}')
+        user_made: Union[str, bool] = data.get('user_made', False)
+        if not isinstance(user_made, bool):
+            try:
+                user_made = bool(user_made)
+            except Exception:
+                raise OrchestratorError(f'Expected user_made to be bool but got {type(user_made)}')
+        return cls(cert=cert, user_made=user_made)
+
+
+class PrivKey():
+    def __init__(self, key: str = '', user_made: bool = False) -> None:
+        self.key = key
+        self.user_made = user_made
+
+    def __bool__(self) -> bool:
+        if self.key:
+            return True
+        return False
+
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, PrivKey):
+            return self.key == other.key and self.user_made == other.user_made
+        return NotImplemented
+
+    def to_json(self) -> Dict[str, Union[str, bool]]:
+        return {
+            'key': self.key,
+            'user_made': self.user_made
+        }
+
+    @classmethod
+    def from_json(cls, data: Dict[str, str]) -> 'PrivKey':
+        if 'key' not in data:
+            return cls()
+        key = data['key']
+        if not isinstance(key, str):
+            raise OrchestratorError('Tried to make PrivKey object with non-string key')
+        if any(k not in ['key', 'user_made'] for k in data.keys()):
+            raise OrchestratorError(f'Got unknown field for PrivKey object. Fields: {data.keys()}')
+        user_made: Union[str, bool] = data.get('user_made', False)
+        if not isinstance(user_made, bool):
+            try:
+                user_made = bool(user_made)
+            except Exception:
+                raise OrchestratorError(f'Expected user_made to be bool but got {type(user_made)}')
+        return cls(key=key, user_made=user_made)
+
+
+class CertKeyStore():
+    service_name_cert = [
+        'rgw_frontend_ssl_cert',
+        'iscsi_ssl_cert',
+        'ingress_ssl_cert',
+    ]
+
+    host_cert = [
+        'grafana_cert',
+        'alertmanager_cert',
+        'prometheus_cert',
+        'node_exporter_cert',
+    ]
+
+    host_key = [
+        'grafana_key',
+        'alertmanager_key',
+        'prometheus_key',
+        'node_exporter_key',
+    ]
+
+    service_name_key = [
+        'iscsi_ssl_key',
+        'ingress_ssl_key',
+    ]
```

Comment:
You can't move all of these to the spec classes. The ingress, rgw, and iscsi related ones can, but prometheus, grafana, etc. were never part of a spec. We just generate those certs at misc. spots in the codebase and then were either storing them in the config-key store or just not storing them at all.

Reply:
@adk3798 I see your point .. monitoring has always been treated differently which is a problem. Can't we use the class `class MonitoringSpec` to hold this info?

Reply:
in theory, yes, but that would be a pretty big configuration change to move those certs into the spec. It might go okay, although I think we'd need to have a spec for each monitoring daemon type (if they don't already exist. I think most, if not all, do). I'm open to the idea. Maybe a topic for the weekly.